### PR TITLE
Feature - apply wiki data

### PR DESCRIPTION
**Pull request by fabricantes**

### CODE REVIEW

File: pepdb/core/migrations/0154_auto_20181018_1653.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('core', '0153_auto_20180911_1325'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='person',
+            name='en_wiki_url',
```
Comment (dchaplinsky) [inline pepdb/core/migrations/0154_auto_20181018_1653.py]: No, mate, you should use django translations. Check core/translation.py

File: pepdb/tasks/management/commands/apply_wiki_data.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        self.dob_updated = 0
+        self.photo_updated = 0
+
+        activate(settings.LANGUAGE_CODE)
+
+        wiki_matches = WikiMatch.objects.filter(status="a")
+
+        with tqdm.tqdm(total=wiki_matches.count()) as pbar:
+
+            for match in wiki_matches.select_related("person").nocache().iterator():
```
Comment (dchaplinsky) [inline pepdb/tasks/management/commands/apply_wiki_data.py]: You probably can wrap wiki_matches.select_related("person").nocache().iterator() into tqdm.tqdm() and don't use pbar.update

File: pepdb/tasks/management/commands/apply_wiki_data.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+        person_photo = match.person.photo
+
+        if person_photo or not wikidata_photo_name:
+            return
+
+        wikidata_photo_name = wikidata_photo_name.replace(" ", "_")
+        md5 = hashlib.md5()
+        md5.update(wikidata_photo_name.encode("utf-8"))
+        hash = md5.hexdigest()
```
Comment (dchaplinsky) [inline pepdb/tasks/management/commands/apply_wiki_data.py]: hash is a system keyword

File: pepdb/tasks/management/commands/apply_wiki_data.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        if not wikidata_dob_obj:
+            return
+
+        wikidata_dob = self.parse_wikidata_dob(wikidata_dob_obj)
+
+        if not wikidata_dob:
+            return
+
+        if not person_dob:
+            match.person.dob = wikidata_dob
```
Comment (dchaplinsky) [inline pepdb/tasks/management/commands/apply_wiki_data.py]: What about cases when only year is available in wikidata?

Comment (fabricantes) [reply pepdb/tasks/management/commands/apply_wiki_data.py]: Thought we only need full dob from this source. Will match partial as well

File: pepdb/tasks/management/commands/apply_wiki_data.py (hunk 3 above)
Comment (fabricantes) [reply pepdb/tasks/management/commands/apply_wiki_data.py]: true

**Comment by dchaplinsky** [comment]
Ok, here goes solution for UA/EN wiki pages.

For UA we are trying to put links in following sequence (whichever available first):
UA/RU/EN
For EN:
EN/UA/RU

**Comment by fabricantes** [comment]
> Ok, here goes solution for UA/EN wiki pages.
> 
> For UA we are trying to put links in following sequence (whichever available first):
> UA/RU/EN
> For EN:
> EN/UA/RU

Ok. This has been done.

**Pull request merged**